src/components/Shared/MetalSummary.js

```diff
@@ -0,0 +1,132 @@
+import { useContext, useState, useEffect } from 'react'
+import Table from 'src/components/Shared/Table'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { useResourceQuery } from 'src/hooks/resource'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { InventoryRepository } from 'src/repositories/InventoryRepository'
+
+const MetalSummary = ({ originalList }) => {
+  const { getRequest } = useContext(RequestsContext)
+  const [gridData, setGridData] = useState({})
+
+  async function fetchGridData() {
+    var parameters = ``
+    getRequest({
+      extension: InventoryRepository.Metals.qry,
+      parameters: parameters
+    })
+      .then(res => {
+        const metalsStoreList = res.list
+
+        const updatedData = originalList.map(item => {
+          const matchedMetal = metalsStoreList.find(metalStoreItem => item.metalId === metalStoreItem.recordId)
+
+          if (matchedMetal) {
+            return { ...item, metalRef: matchedMetal.reference }
+          }
+
+          return item
+        })
+
+        const viewedList = processMetalsList(updatedData)
+
+        const transformedData = {
+          count: viewedList.length,
+          list: viewedList
+        }
+
+        setGridData(transformedData)
+      })
+      .catch(error => {})
+  }
+
+  const processMetalsList = metalsList => {
+    // Update weight based on priceType
```

Comment:
remove comments

Reply:
i kept them on purpose to have clear logic